firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/datagrams.txt

```diff
@@ -0,0 +1,446 @@
+      THEN("After the transform method is called, the value returned by the output_datagram's seq accessor method is equal to the sequence field of the input_buffer body's header")
+      THEN("The value returned by the output_datagram's length accessor method is equal to the length field of the input_buffer body's header")
+      THEN("The value returned by the output_datagram's length accessor method is inconsistent with the actual length of the payload buffer from the body")
+      THEN("The paylaod returned by the output_datagram's paylaod accessor method is equal to the paylaod from the input_buffer body")
+      THEN("The output_datagram's payload accessor method returns a reference to the payload given in its constructor")
+      THEN("The data in the output_datagram's payload buffer is independent of the data in input_buffer")
+
+    WHEN("A body with a non-empty paylaod buffer and a complete 2-byte header where the sequence field of the header does not match sequence given in output_datagram's constructor")
+      THEN("The transform method reports invalid sequence status")
+      THEN("After the transform method is called, the value returned by the output_datagram's seq accessor method is equal to the sequence field of the input_buffer body's header")
+      THEN("The value returned by the output_datagram's length accessor method is equal to the length field of the input_buffer body's header")
+      THEN("The paylaod returned by the output_datagram's paylaod accessor method is equal to the paylaod from the input_buffer body")
+      THEN("The output_datagram's payload accessor method returns a reference to the payload given in its constructor")
+      THEN("The data in the output_datagram's payload buffer is independent of the data in input_buffer")
+
+    WHEN("A body with a complete 2-byte header, a non-empty payload buffer consistent with the length field of the header is given to the receiver")
+      THEN("The transform method reports ok status")
+      THEN("After the transform method is called, the value returned by the output_datagram's seq accessor method is equal to the sequence field of the input_buffer body's header")
+      THEN("The value returned by the output_datagram's length accessor method is equal to the length field of the input_buffer body's header")
+      THEN("The paylaod returned by the output_datagram's paylaod accessor method is equal to the paylaod from the input_buffer body")
+      THEN("The output_datagram's payload accessor method returns a reference to the payload given in its constructor")
+      THEN("The data in the output_datagram's payload buffer is independent of the data in input_buffer")
+
+Scenario: Datagram Receiver correctly reports incrementing, rollover, and resetting of expected sequence number
+
+  GIVEN("A Datagram receiver of buffer size 254 bytes and expected sequence number equal to 1")      
+  // assertions carried out in GIVEN:
```

Comment:
These assertions need to be given more context in order to make sense, because output_datagram isn't declared in the GIVEN statement. If these are just describing the first input_buffer you call transform on, and you want to describe what's happening in setup, maybe it's enough to just say something like the following:
```
// Setup:
// Call transform with an input_buffer where the header's seq field has value 0
```